src/Core/Generator/SchemaGeneratorFactory.cs

```diff
@@ -56,46 +58,78 @@ public static async Task<string> Create(RuntimeConfig config, string mode, int?
-            SamplingModes samplingMode = (SamplingModes)Enum.Parse(typeof(SamplingModes), mode);
+            foreach (KeyValuePair<string, Entity> entity in config.Entities)
+            {
+                dbAndContainerToProcess.Add(entity.Value.Source.Object);
+            }
 
-            // Determine the appropriate sampler based on the sampling mode.
-            ISchemaGeneratorSampler schemaGeneratorSampler = samplingMode switch
+            StringBuilder schema = new();
+            foreach (string dbAndContainer in dbAndContainerToProcess)
             {
-                SamplingModes.TopNExtractor => new TopNExtractor(container, sampleCount, days, logger),
-                SamplingModes.EligibleDataSampler => new EligibleDataSampler(container, partitionKeyPath, sampleCount, days, logger),
-                SamplingModes.TimePartitionedSampler => new TimePartitionedSampler(container, groupCount, sampleCount, days, logger),
-                _ => throw new ArgumentException($"Invalid sampling mode: {mode}, Valid Sampling Modes are: {SamplingModes.TopNExtractor}, {SamplingModes.EligibleDataSampler}, {SamplingModes.TimePartitionedSampler}")
-            };
+                if (string.IsNullOrEmpty(dbAndContainer)) // Safety check
+                {
+                    continue;
+                }
 
-            logger.LogInformation("Sampling Started using {0}", schemaGeneratorSampler.GetType().Name);
+                string[] dbContainer = dbAndContainer.Split('.');
+                if (dbContainer.Length != 2)
+                {
+                    logger.LogError("Invalid Database and Container Name provided in the config file");
+                    throw new ArgumentException("Invalid Database and Container Name provided in the config file");
+                }
 
-            // Get sample data from the selected sampler.
-            List<JsonDocument> dataArray = await schemaGeneratorSampler.GetSampleAsync();
+                logger.LogInformation("Connecting to Cosmos DB Database: {0}, Container: {1}", dbContainer[0], dbContainer[1]);
 
-            logger.LogInformation("{0} records collected as Sample", dataArray.Count);
-            if (dataArray.Count == 0)
-            {
-                logger.LogError("No data was sampled. Please try a different sampling mode or provide different sampling options.");
-                throw new ArgumentException("No data was sampled. Please try a different sampling mode or provide different sampling options.");
-            }
+                // Connect to the Azure Cosmos DB container.
+                container ??= ConnectToCosmosDB(connectionString, dbContainer[0], dbContainer[1]);
+                SamplingModes samplingMode = (SamplingModes)Enum.Parse(typeof(SamplingModes), mode);
+
+                // Determine the appropriate sampler based on the sampling mode.
+                ISchemaGeneratorSampler schemaGeneratorSampler = samplingMode switch
```

Comment:
we can move some logic out of this like getSampler

Reply:
I believe its fine, its not too many lines of code for get the sampler